evalml/tests/data_checks_tests/test_id_columns_data_check.py

```diff
@@ -293,17 +293,24 @@ def test_unidentified_first_col_primary_key(
     )
 
     id_cols_check = IDColumnsDataCheck(id_threshold=0.95)
+    if input_type == "string":
+        order = ["col_2", "col_3_id", "col_1_id"]
+        order_msg = "Columns 'col_2', 'col_3_id', 'col_1_id' are 95.0% or more likely to be an ID column"
+    else:
+        order = ["col_2", "col_1_id", "col_3_id"]
+        order_msg = "Columns 'col_2', 'col_1_id', 'col_3_id' are 95.0% or more likely to be an ID column"
+
```

Comment:
What change caused this? It's a surprising distinction to make

Also, a nitpicky clarity change - we can replace the separate order messages here with `order_msg = f"Columns '{order[0]}', '{order[1]}', '{order[2]}' are 95% or more..."` just to make it clearer what the difference is!

Reply:
In this case, `col_3_id` is a random series of string-represented numbers, which under the new inference will become the `Integer` `ltype`. Following [this](https://github.com/alteryx/evalml/blob/main/evalml/data_checks/id_columns_data_check.py#L178) logic in validate, since they're all unique values and since the column ends in `id`, it gets identified as a possible ID column.

Previously however, since this column wasn't a numeric logical type, the logic [here](https://github.com/alteryx/evalml/blob/main/evalml/data_checks/id_columns_data_check.py#L178) didn't catch it, but [this](https://github.com/alteryx/evalml/blob/main/evalml/data_checks/id_columns_data_check.py#L203) did.

This just ends up resulting in a different ordering by which these columns are caught as possible ID columns by the data check